### ISSUE TITLE
onError not called during chatManager.connect(...)

### ISSUE THREAD

**Issue by mikebridge** [pusher/chatkit-client-js#8]
My `onError` callback doesn't seem to be called when I call `connect()`.  The logger is working, though:

```typescript

const logger = {
    verbose: (message: string, error?: Error) => {
        console.log(`VERBOSE: ${message}`);
        console.log(error);
    },
    debug: (message: string, error?: Error) => {
        console.log(`DEBUG: ${message}`);
        console.log(error);
    },
    info: (message: string, error?: Error) => {
        console.log(`INFO: ${message}`);
        console.log(error);
    },
    warn: (message: string, error?: Error) => {
        console.log(`WARN: ${message}`);
        console.log(error);
    },
    error: (message: string, error?: Error) => {
        console.log(`ERROR: ${message}`);
        console.log(error);
    }
};

const onSuccess = currentUser => {
    console.log("Successful connection", currentUser);
},

const onError = error => {
    console.log("Error on connection");
    console.error(error);
};

const chatManager = new ChatKit.ChatManager({       
    instanceLocator,
    tokenProvider,
    logger
});

chatManager.connect({onSuccess, onError});

```

```
Connecting to v1:us1:xxxxx
withChatKit.tsx:81 VERBOSE: RetryingSubscription: transitioning to OpeningSubscriptionState
withChatKit.tsx:82 undefined
withChatKit.tsx:81 VERBOSE: TokenProvidingSubscription: transitioning to TokenProvidingState
withChatKit.tsx:82 undefined
withChatKit.tsx:81 VERBOSE: TokenProvidingSubscription: token fetched: ey xxxxx
withChatKit.tsx:82 undefined
withChatKit.tsx:81 VERBOSE: TokenProvidingSubscription: transitioning to FailedSubscriptionState
withChatKit.tsx:82 ErrorResponse {statusCode: 404, headers: {…}, info: {…}}
withChatKit.tsx:81 VERBOSE: RetryingSubscription: transitioning to RetryingSubscriptionState
withChatKit.tsx:82 undefined
withChatKit.tsx:81 VERBOSE: RetryResolution:  Error received
withChatKit.tsx:82 ErrorResponse {statusCode: 404, headers: {…}, info: {…}}
withChatKit.tsx:81 VERBOSE: RetryResolution: Error is not retryable
withChatKit.tsx:82 ErrorResponse {statusCode: 404, headers: {…}, info: {…}}
withChatKit.tsx:81 VERBOSE: RetryingSubscription: transitioning to FailedSubscriptionState
withChatKit.tsx:82 ErrorResponse {statusCode: 404, headers: {…}, info: {…}}
```

I would have expected to see an "error on connection" log via my onError callback.  (the actual error is that my user hasn't been created yet.)

**Comment by hamchapman** [comment]
Thanks for the report - I'll take a look into this today 👍

**Comment by hamchapman** [comment]
I've released 0.2.0 which fixes this 👍
